src/databricks/labs/ucx/hive_metastore/locations.py

```diff
@@ -248,3 +250,109 @@ def _try_fetch(self) -> Iterable[Mount]:
             f"SELECT * FROM {escape_sql_identifier(self._schema)}.{escape_sql_identifier(self._table)}"
         ):
             yield Mount(*row)
+
+
+@dataclass
+class TableInMount:
+    format: str
+    is_partitioned: bool
+
+
+class TablesInMounts(CrawlerBase[Table]):
+
+    def __init__(
+        self,
+        backend: SqlBackend,
+        ws: WorkspaceClient,
+        inventory_database: str,
+        mc: Mounts,
+        include_mounts: list[str] | None = None,
+        filter_paths_in_mount: list[str] | None = None,
+    ):
+        super().__init__(backend, "hive_metastore", inventory_database, "tables", Table)
+        self._dbutils = ws.dbutils
+        self._mc = mc
+        self._include_mounts = include_mounts
+
+        irrelevant_patterns = {'_SUCCESS', '_committed_', '_started_'}
+        if filter_paths_in_mount:
+            irrelevant_patterns.update(filter_paths_in_mount)
+        self._fiter_paths = irrelevant_patterns
+
+    def snapshot(self) -> list[Table]:
+        """
+        Takes a snapshot of tables in the specified catalog and database.
+
+        Returns:
+            list[Table]: A list of Table objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
+
+    def _try_load(self) -> Iterable[Table]:
+        """Tries to load table information from the database or throws TABLE_OR_VIEW_NOT_FOUND error"""
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Table(*row)
+
+    def _crawl(self):
+        all_mounts = self._mc.snapshot()
+        all_tables = []
+        for mount in all_mounts:
+            if self._include_mounts and mount.name not in self._include_mounts:
+                continue
+            table_paths = self._find_delta_log_folders(mount.name)
+            for path, entry in table_paths.items():
+                guess_table = os.path.basename(path)
+                table = Table(
+                    catalog="hive_metastore",
+                    database=f"mounted_{mount.name}",
+                    name=guess_table,
+                    object_type="EXTERNAL",
+                    table_format=entry.format,
+                    location=path,
+                    storage_properties="PARTITIONED" if entry.is_partitioned else None,
+                )
+                all_tables.append(table)
+        return all_tables
+
+    def _find_delta_log_folders(self, root_dir, delta_log_folders=None) -> dict:
```

Comment:
Type hints requires. if delta_log_folders is dict, it can be initialized with default value as empty dict.